velox/functions/lib/string/StringImpl.h

```diff
@@ -688,4 +689,113 @@ FOLLY_ALWAYS_INLINE void pad(
       padPrefixByteLength);
 }
 
+namespace detail {
+
+template <bool forSpark>
+inline bool isStrictSpaceAscii(unsigned char ch) {
+  if constexpr (forSpark) {
+    return ch == ' ';
+  } else {
+    return std::isspace(ch);
+  }
+}
+
+template <bool forSpark>
+inline bool isStrictSpaceUtf8(utf8proc_int32_t cp) {
+  if constexpr (forSpark) {
+    return cp == 0x20;
+  } else {
+    return isUnicodeWhiteSpace(cp);
+  }
+}
+
+// ASCII InitCap Implementation.
+template <bool forSpark, typename TOutString, typename TInString>
+FOLLY_ALWAYS_INLINE void initcapAsciiImpl(
+    TOutString& output,
+    const TInString& input) {
+  output.resize(input.size());
+  const char* inputChars = input.data();
+  char* outputChars = output.data();
+
+  bool isStartOfWord = true;
+  for (size_t i = 0; i < input.size(); ++i) {
+    unsigned char currentChar = static_cast<unsigned char>(inputChars[i]);
+
+    if (isStrictSpaceAscii<forSpark>(currentChar)) {
+      isStartOfWord = true;
+      outputChars[i] = currentChar;
+    } else if (isStartOfWord) {
+      outputChars[i] = std::toupper(currentChar);
+      isStartOfWord = false;
+    } else {
+      outputChars[i] = std::tolower(currentChar);
+    }
+  }
+}
+
+// UTF-8 InitCap Implementation.
+template <bool forSpark, typename TOutString, typename TInString>
+FOLLY_ALWAYS_INLINE bool initcapUtf8Impl(
+    TOutString& output,
+    const TInString& input) {
+  const uint8_t* inputBytes = reinterpret_cast<const uint8_t*>(input.data());
+  const uint8_t* inputEnd = inputBytes + input.size();
```

Comment:
Use `const char*` for `inputBytes` and `inputEnd` to avoid reinterpret_cast at L769 and L778.